# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.cpp
# Commit: f06186536887
# Full Hash: f061865368870d7e33c22d537992d272916ebec1
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-10-03 20:53 +0000
# Regressor Bug: 1968890
# File Overlap Count: 3
# Overlapping Files: dom/worklet/WorkletFetchHandler.cpp, dom/workers/ScriptLoader.cpp, dom/script/ScriptLoader.cpp
# Description:
#   Bug 1968890 - Part 6: ScriptLoaders update. r=jonco,dom-worker-reviewers,edenchuang
# ==============================================================================

diff -r 584beaae1d20 -r f06186536887 dom/workers/ScriptLoader.cpp
--- a/dom/workers/ScriptLoader.cpp	Fri Oct 03 20:53:29 2025 +0000
+++ b/dom/workers/ScriptLoader.cpp	Fri Oct 03 20:53:29 2025 +0000
@@ -1730,16 +1730,8 @@
   moduleRequest->OnFetchComplete(loadContext->mLoadResult);
 
   if (NS_FAILED(loadContext->mLoadResult)) {
-    if (moduleRequest->IsDynamicImport()) {
-      if (request->isInList()) {
-        moduleRequest->CancelDynamicImport(loadContext->mLoadResult);
-        mScriptLoader->TryShutdown();
-      }
-    } else if (!moduleRequest->IsTopLevel()) {
-      moduleRequest->Cancel();
+    if (moduleRequest->IsDynamicImport() || !moduleRequest->IsTopLevel()) {
       mScriptLoader->TryShutdown();
-    } else {
-      moduleRequest->LoadFailed();
     }
   }
   return true;